### PR TITLE
Arrow Patch

### LINKED ISSUE #267: Failing to make sound when getting hit by an arrow with null shooter
### Observed Behavior

Whenever an arrow summoned with /summon hits the ground/a player, no sound effect is made and there's a stacktrace in console.
```
[15:34:57 WARN]: Entity threw exception at world:81.96561319573252,70.74121830390679,258.8281078198744
[15:34:57 WARN]: java.lang.NullPointerException: Cannot read field "locX" because "entity" is null
[15:34:57 WARN]:        at net.minecraft.server.v1_8_R3.World.makeSound(World.java:1171)
[15:34:57 WARN]:        at net.minecraft.server.v1_8_R3.EntityArrow.t_(EntityArrow.java:326)
[15:34:57 WARN]:        at net.minecraft.server.v1_8_R3.World.entityJoinedWorld(World.java:1884)
[15:34:57 WARN]:        at net.minecraft.server.v1_8_R3.World.g(World.java:1851)
[15:34:57 WARN]:        at net.minecraft.server.v1_8_R3.World.tickEntities(World.java:1677)
[15:34:57 WARN]:        at net.minecraft.server.v1_8_R3.WorldServer.tickEntities(WorldServer.java:602)
[15:34:57 WARN]:        at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:863)
[15:34:57 WARN]:        at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:409)
[15:34:57 WARN]:        at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:724)
[15:34:57 WARN]:        at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:617)
[15:34:57 WARN]:        at java.base/java.lang.Thread.run(Thread.java:833)
```
### Expected Behavior

The sound effect plays.

### Steps To Reproduce

1. Join the server.
2. Summon an arrow with /summon

### Plugin List

_No response_

### Server Version

```
>version
[14:55:58 INFO]: This server is running NachoSpigot version git-NachoSpigot-"06a9e2a" (MC: 1.8.8) (Implementing API version 1.8.8-R0.2-SNAPSHOT)
[14:55:58 INFO]: You are running the latest version
```

### Other

_No response_

### Agreements

- [X] You were able to find this issue on the latest version of NachoSpigot.
- [X] You have confirmed that there aren’t any issues open regarding this bug.
- [X] You have confirmed that you are NOT using a fork of NachoSpigot in any way. **YOUR changes are not OUR faults.**

### PR DESCRIPTION
**Pull request by Tofpu**
# Description

Please provide a detailed description of the changes you have made.

Fixes/Adds # (issue)
- Fixes #267

How has this been tested?
- By spawning the arrow with the `/summon` command

How have you tested the changes you have made?
- I used an alternating playSound method, which uses the x y z parameters instead of the shooter entity

# Checklist:

- [x] I have reviewed my code thoroughly.
- [x] I have tested my code.
- [x] My changes generate no new warnings

### CODE REVIEW

File: NachoSpigot-Server/src/main/java/net/minecraft/server/EntityArrow.java [hunk 1]
```diff
@@ -323,7 +323,7 @@ public void t_() {
                     this.locX -= this.motX / (double) f1 * 0.05000000074505806D;
                     this.locY -= this.motY / (double) f1 * 0.05000000074505806D;
                     this.locZ -= this.motZ / (double) f1 * 0.05000000074505806D;
-                    world.makeSound(this.shooter,"random.bowhit", 1.0F, 1.2F / (this.random.nextFloat() * 0.2F + 0.9F));
+                    world.makeSound(d, e, f,"random.bowhit", 1.0F, 1.2F / (this.random.nextFloat() * 0.2F + 0.9F));
```
[the remark this answers is not in the document]
Comment (Tofpu) [reply NachoSpigot-Server/src/main/java/net/minecraft/server/EntityArrow.java:326]: Good catch!

**Comment by Elierrr** [comment]
Still receiving the error when the arrow hits the ground instead of a player, you'll have to do the same for line 297.﻿

**Comment by Tofpu** [comment]
> Still receiving the error when the arrow hits the ground instead of a player, you'll have to do the same for line 297.
Got it! Could you by any chance tell me what you did to get that error?

**Comment by Elierrr** [comment]
> when the arrow hits the ground instead of a player,

**Comment by Tofpu** [comment]
> ~> when the arrow hits the ground instead of a player,~ When the arrow hits a player.

Aha, that edit makes sense now. I appreciate the pointers!

**Comment by Elierrr** [comment]
When it hits a player there is no error in console however there still is no sound effect.﻿

**Comment by Tofpu** [comment]
> When it hits a player there is no error in console however there still is no sound effect.

Hmm, I thought that was to be intended lol, anyway I checked the code further and it seems like the d, e and f values are not being updated, since an entity was involved.

I'm not very familiar with the server's codebase but I'm thinking providing the involved entity would do it. I'll test the theory in a few moments

**Review by Elierrr — APPROVED** [review]
Tested and works as expected!﻿

**Comment by Lucaskyy** [comment]
Thanks, Tofpu for this PR! :)

**Pull request merged**